htdocs/modules/system/admin/preferences.php

```diff
@@ -369,7 +369,13 @@
 
 				case 'select_geshi' :
 					$ele = new icms_form_elements_Select($title, $config[$i]->getVar('conf_name'), $config[$i]->getConfValueForOutput());
-					$dirlist = str_replace('.php', '', icms_core_Filesystem::getFileList(ICMS_LIBRARIES_PATH . '/geshi/geshi/', '', array('php')));
+					$reflector = new ReflectionClass("\\GeSHi");
```

Comment:
Really good idea to use reflection, I was having a similar issue with loading the Smarty config : in essence you don't know where the library will be put on the file system, composer makes that transparent. So you cannot assume that your fiels will be in a specific spot.

Reply:
Yeah that would be good idea, but problem here is that in Geshi language files are not classes but only PHP arrays, so it's impossible to use such data in reflection.

Reply:
Thanks! 

...and oh, I wrote previously here comment not about what it is... :(